core/src/server/rpc_module.rs

```diff
@@ -689,39 +689,14 @@ impl<Context: Send + Sync + 'static> RpcModule<Context> {
 			return Err(Error::SubscriptionNameConflict(subscribe_method_name.into()));
 		}
 
-		self.methods.verify_method_name(subscribe_method_name)?;
```

Comment:
can you explain this? 

these are still required AFAIU, if the method name is already registered we should still fire an error

Reply:
The problem is that you do check for the subscription and unsubscription separately.
Thus, if only the unsubscription fails we should still deny both.

For example, if the subscription is successful and the unsubscription fails you can't really unsubscribe :thinking:

Reply:
I see now what you mean! Indeed this was living the door open for having the `unsubscriptions` left initialized, have modified to use the previous way of first verifying the method's name. Good catch! Thanks!